tests/unit/amo/components/TestMoreAddonsByAuthorsCard.js

```diff
@@ -0,0 +1,456 @@
+  });
+
+  it('should dispatch a fetch action if no add-ons found', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandler,
+      store,
+    });
+
+    sinon.assert.calledWith(dispatchSpy, fetchAddonsByAuthors({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandlerId: errorHandler.id,
+    }));
+  });
+
+  it('should dispatch a fetch action if props are updated', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandler,
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    root.setProps({
+      addonType: ADDON_TYPE_THEME,
+      authorNames: ['test1'],
+    });
+
+    sinon.assert.calledWith(dispatchSpy, fetchAddonsByAuthors({
+      addonType: ADDON_TYPE_THEME,
+      authorNames: ['test1'],
+      errorHandlerId: errorHandler.id,
+    }));
+
+    // Make sure an authorNames update even with the same addonType dispatches
+    // a fetch action.
+    dispatchSpy.reset();
+
+    root.setProps({
+      addonType: ADDON_TYPE_THEME,
+      authorNames: ['test2'],
+    });
+
+    sinon.assert.calledWith(dispatchSpy, fetchAddonsByAuthors({
+      addonType: ADDON_TYPE_THEME,
+      authorNames: ['test2'],
+      errorHandlerId: errorHandler.id,
+    }));
+  });
+
+  it('should dispatch a fetch action if forAddonSlug is updated', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandler,
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    root.setProps({
+      forAddonSlug: 'testing',
+    });
+
+    sinon.assert.calledWith(dispatchSpy, fetchAddonsByAuthors({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandlerId: errorHandler.id,
+      forAddonSlug: 'testing',
+    }));
+  });
+
+  it('should not dispatch a fetch action if props are not changed', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = render({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+      errorHandler,
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    root.setProps({
+      addonType: ADDON_TYPE_EXTENSION,
+      authorNames: ['test2'],
+    });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('shows dictionaries in header for ADDON_TYPE_DICT', () => {
```

Comment:
There is a lot of repetition in the following 10 tests. It seems like a test helper method could be used to eliminate a lot of this repetition and reduce the cases to just calling a helper with particular arguments and then including a single assertion.